src/AutoRest.CSharp.V3/Generation/Writers/JsonCodeWriterExtensions.cs

```diff
@@ -17,9 +17,6 @@ public static void ToSerializeCall(this CodeWriter writer, JsonSerialization ser
         {
             writerName ??= w => w.AppendRaw("writer");
 
-            var type = serialization.Type;
-            CSharpType implementationType = typeFactory.CreateType(type);
```

Comment:
The type factory requires schemas created from the swagger. We make dynamic data (property + schema) for flattening. This means the factory won't be able to resolve that, so we can't use it in serialization.

The good news is that this data was redundant (in both serialization and deserialization). It is only used for nullability and if it is a value type. We know this data already, so I use the data that already exists.

Reply:
I'd like to keep the original logic now that it's possible, we are going to have struct models soon.

Reply:
Because of the changes we just made, we cannot always create a type (as it may not have one). We've worked around this by making the logic below recognize if we have a type or not.

Reply:
Are you sure? I think all places you've changed have provably not-null type.

Reply:
Fixed. We're now using the CSharpType IsNullable instead of the TypeReference one. Also reverted the overload for AppendNullableValue.